### PR TITLE
Adding initial support to sycl 1.2.1

### PR DESCRIPTION
**Pull request by marcelsheeny**
Added a patch to initial support to sycl 1.2.1.
pyramid.cpp and simple_conv.cpp examples were removed since it is broken in this current version.

### CODE REVIEW

File: cmake/FindComputeCpp.cmake [hunk 1]
```diff
@@ -118,7 +118,7 @@ else()
   message(STATUS "compute++ flags - ${COMPUTECPP_DEVICE_COMPILER_FLAGS}")
 endif()
 
-set(COMPUTECPP_DEVICE_COMPILER_FLAGS ${COMPUTECPP_DEVICE_COMPILER_FLAGS} -sycl-compress-name -no-serial-memop)
+set(COMPUTECPP_DEVICE_COMPILER_FLAGS ${COMPUTECPP_DEVICE_COMPILER_FLAGS} -sycl-compress-name -no-serial-memop -Wno-ignored-qualifiers)
```
Comment (DuncanMcBain) [inline cmake/FindComputeCpp.cmake]: I'd added this to make the compilation less noisy - it's not a necessary flag!

**Comment by Ruyk** [comment]
why are the examples broken? is it a compile time error due to the changes in the interface or is there some bug in ComputeCpp? can you elaborate?

**Comment by marcelsheeny** [comment]
There were some operator overloading functions in the include/pixel/pixel.hpp file which are giving conflicts (compilation error) with some built-in sycl operator functions. I removed some of those functions and now it is using the default sycl functions.

The broken examples are being compiled, however giving wrong results. Those examples use the convolution operation. When I run make test, I can see that those operations are failing

**Comment by Ruyk** [comment]
Can you disable them using a macro instead? I guess when @mehdi-goli  is back from holidays he can take a look to the problem.

**Comment by marcelsheeny** [comment]
I added the macro to disable the pyramid and simple_conv examples.

**Comment by Ruyk** [comment]
Looks good to me, squash and merge please!

**Pull request merged**